packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -67,24 +67,29 @@ contract SgxVerifier is EssentialContract, IVerifier {
 
     /// @notice Adds trusted SGX instances to the registry.
     /// @param _instances The address array of trusted SGX instances.
-    function addInstances(address[] calldata _instances) external onlyOwner {
+    /// @return ids The respective instanceId array per addresses.
+    function addInstances(address[] calldata _instances)
+        external
+        onlyOwner
+        returns (uint256[] memory ids)
+    {
         if (_instances.length == 0) revert SGX_INVALID_INSTANCES();
 
-        for (uint256 i; i < _instances.length; ++i) {
-            _addInstance(_instances[i]);
-        }
+        ids = _addInstances(_instances, false);
     }
 
     /// @notice Adds SGX instances to the registry by another SGX instance.
     /// @param id The id of the SGX instance who is adding new members.
     /// @param _instances The address array of SGX instances.
     /// @param signature The signature proving authenticity.
+    /// @return ids The respective instanceId array per addresses.
```

Comment:
If we always need a new instance to replace the old one, and an array of extra instances to be added, maybe it make sens to use two parameters to replace this current on `_instances`.

Reply:
You mean to separate them ? (` address myNewInstance, address [] newInstancesToBeAdded`)